### PR TITLE
Finish member sign up page and welcoming

### PR DESCRIPTION
**Pull request by bobmayuze**
Finish member sign up page and welcoming

### CODE REVIEW

File: src/web/src/components/SignUp.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  },
+  methods: {
+    onSubmit(evt) {
+      evt.preventDefault();
+      // alert(JSON.stringify(this.form));
+      signup(this.form).then((response)=>{
+        console.log(response);
+        if (response.data.content){
+          alert(response.data.content['msg'])
+          alert('Please go to log in')
```
Comment (123joshuawu) [inline src/web/src/components/SignUp.vue:76]: Can use BootstrapVue toast notifications for this

File: src/web/src/components/Header.vue [hunk 2]
```diff
@@ -72,19 +82,26 @@ import { getDefaultSemester } from '@/services/AdminService';
...
+      SignUpForm : SignUpComponent
+    },
     data() {
       return {
         form: {
           email: '',
           password: '',
         },
         isLoggedIn: false,
+        showSingUp: false,
```
Comment (123joshuawu) [inline src/web/src/components/Header.vue:102]: er *`showSignUp`

Comment (bobmayuze) [reply src/web/src/components/Header.vue:102]: lol

Comment (jshom) [reply src/web/src/components/Header.vue:102]: @bobmayuze can u finish this up when u get a chance, thx

**Comment by marchdan** [comment]
@bobmayuze Can you add a testing procedure to the original comment from PR creation so we can properly test this?

**Comment by bobmayuze** [comment]
Test Instruction
1. run ./scripts/dev-start-legacy.sh
2. go to localhost:8080
3. Click top right corner, the user login/signup window should pop up
4. click the sign-up button, and fill out the form
5. click submit, u should see an alert window says "user successfully created" 
6. click 'go to login'
7. use the email and password to log in
8. top right corner should say "HI {your name}"

**Comment by jshom** [comment]
**Changes**

- [ ] Only required fields should be: Name, Email, Password, grad/undergrad and major leave as optional fields. Remove phonenumber from form. Lets make it easy to use YACS.
- [ ] Add sign/up button to header, should be one click and form to make an account. No need to channel through login
- [ ] Remove "go back to log-in" redundant, exists "X" at the top right
- [ ] Remove clear button, not that much data to punch in/edit manually
- [ ] Rename sign up form to "Sign Up" from "Login/Sign-up"
- [ ] Rename "Your Name" to "Full Name", clearer
- [ ] Automatically log in user and reload page after submit, make it easy to get going after signup. Lets not use the alerts to inform login
- [ ] Comment out profile option in user dropdown, we can add later
---
(photo for info about form edits)
Screen Shot 2020-04-21 at 7 02 06 PM

(In list of tasks) better to get reloaded and have auto login than generic js popup/alert
Screen Shot 2020-04-21 at 7 02 33 PM

(User dropdown)
Screen Shot 2020-04-21 at 7 16 18 PM

**Comment by jshom** [comment]
Awesome, good stuff y'all:
**Last few things:**
- [ ] Reorder fields, move name to the top so that order is Full Name, Email, Password, Grad/Undergrad, Major
- [ ] Change login and sign up button styling, have more colored / bold on the right, and lighter on the left. (Sign up on the right)
> See LinkedIn or Twitter here:
> Screen Shot 2020-04-21 at 8 55 46 PM
> Screen Shot 2020-04-21 at 8 55 49 PM
- [ ] Remove user added successfully alert, if page reloads, we know its all good
> Screen Shot 2020-04-21 at 9 00 44 PM
- [ ] Change text on submit button to show "Finish Sign Up!" instead of submit, clearer intent (also add class `btn-block` to occupy full width

**Pull request merged**